src/customization/BlockDecoration.tsx

```diff
@@ -1,27 +1,24 @@
 import React, {Fragment, PropsWithChildren, useContext} from 'react';
 
-import {BlockIdContext} from '../context/blockIdContext';
 import {InnerContext} from '../context/innerContext';
 import {BlockDecorationProps} from '../models';
-import {getBlockIndexFromId} from '../utils';
 
-export const BlockDecoration = (props: PropsWithChildren<BlockDecorationProps>) => {
-    const blockContenxtId = getBlockIndexFromId(useContext(BlockIdContext));
+export const BlockDecoration = ({
+    id,
+    children: blockChildren,
+}: PropsWithChildren<BlockDecorationProps>) => {
     const {headerBlockTypes} = useContext(InnerContext);
-    const blockId = props.id || blockContenxtId;
-    const isHeader = Boolean(props.id && headerBlockTypes.includes(props.id));
+    const isHeader = Boolean(typeof id === 'string' && headerBlockTypes.includes(id));
```

Comment:
Why don't we render header if id === number?

Reply:
it is just flag passed for editor it is not used it constructor itself